agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -69,20 +77,45 @@ public TelemetryItemExporter(
       TelemetryPipeline telemetryPipeline, TelemetryPipelineListener listener) {
     this.telemetryPipeline = telemetryPipeline;
     this.listener = listener;
+    this.attributes = initOtelResourceAttributes();
   }
 
   public CompletableResultCode send(List<TelemetryItem> telemetryItems) {
+    List<List<TelemetryItem>> result =
+        groupTelemetryItemsByConnectionStringAndRoleName(telemetryItems);
+    List<CompletableResultCode> resultCodeList = new ArrayList<>();
+    for (List<TelemetryItem> batch : result) {
+      resultCodeList.add(
+          internalSendByConnectionStringAndRoleName(batch, batch.get(0).getConnectionString()));
+    }
+    return maybeAddToActiveExportResults(resultCodeList);
+  }
+
+  // visible for tests
+  List<List<TelemetryItem>> groupTelemetryItemsByConnectionStringAndRoleName(
+      List<TelemetryItem> telemetryItems) {
     Map<String, List<TelemetryItem>> groupings = new HashMap<>();
+    // group TelemetryItem by connection string
     for (TelemetryItem telemetryItem : telemetryItems) {
       groupings
           .computeIfAbsent(telemetryItem.getConnectionString(), k -> new ArrayList<>())
           .add(telemetryItem);
     }
-    List<CompletableResultCode> resultCodeList = new ArrayList<>();
+
+    // and then group TelemetryItem by role name
+    List<List<TelemetryItem>> result = new ArrayList<>();
     for (Map.Entry<String, List<TelemetryItem>> entry : groupings.entrySet()) {
-      resultCodeList.add(internalSendByConnectionString(entry.getValue(), entry.getKey()));
+      Map<String, List<TelemetryItem>> roleNameGroupings = new HashMap<>();
+      for (TelemetryItem telemetryItem : entry.getValue()) {
+        String roleName = "";
+        if (telemetryItem.getTags() != null) { // Statsbeat doesn't have tags
+          roleName = telemetryItem.getTags().get(ContextTagKeys.AI_CLOUD_ROLE.toString());
```

Comment:
I think we probably need another null check after this line, since telemetry isn't required to have AI_CLOUD_ROLE